### PR TITLE
Bluetooth service

### PR DESCRIPTION
**Pull request by atrovato**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [ ] If your changes modify the API (REST or Node.js), did you modify the documentation? (Documentation is based on comments in code)

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Bluetooth service to handle Bluetooth devices.

### CODE REVIEW

File: server/services/bluetooth/lib/commands/bluetooth.determinePeripheral.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+const { EVENTS, WEBSOCKET_MESSAGE_TYPES } = require('../../../../utils/constants');
+const { connect } = require('../utils/connect.js');
+const { discoverServices } = require('../utils/discoverServices.js');
+const { discoverCharacteristics } = require('../utils/discoverCharacteristics.js');
+const { read } = require('../utils/read.js');
+
+/**
+ * @description Connect to Bluetooth peripheral and read informationto determine which kind of device.
+ * @param {string} uuid - Peripheral UUID.
+ * @example
```
Comment (atrovato) [inline server/services/bluetooth/lib/commands/bluetooth.determinePeripheral.js]: @Pierre-Gilles j'ai une erreur eslint "@returns missing" hors la méthode ne rien, et si j'ajoute le commentaire, j'ai une autre erreur qui me dit justement que la méthode ne retourne rien.
Je pense avoir dévoilé un problème avec eslint.

Comment (atrovato) [reply server/services/bluetooth/lib/commands/bluetooth.determinePeripheral.js]: J'ai du ajouter `/* eslint-disable jsdoc/require-returns */` pour valider le build.
Dans l'attente d'une solution plus propre.

Comment (atrovato) [reply server/services/bluetooth/lib/commands/bluetooth.determinePeripheral.js]: Apparemment, la solution est de faire des méthodes plus courtes, eslint doit être perdu s'il y a trop de lignes.
J'ai pu supprimer le eslint-disabled.

**Comment by atrovato** [comment]
Wait #512 PR to be merged to get device / feature definition updates.

**Comment by Pierre-Gilles** [comment]
Hello! De retour de congés, je reprend les PR une par une.

A quel niveau es-tu sur celle-ci?

**Comment by atrovato** [comment]
Salut, suite à notre discussion sur MQTT et les sub-services, ce service nécessite de grandes modifications. Du coup, la relecture n'est pas encore nécessaire.

**Comment by atrovato** [comment]
The service is still in standby, the community is moving on noble library, and some others may appear soon.
I'll try new version as there are available.

**Comment by Pierre-Gilles** [comment]
Ok!

**Comment by Pierre-Gilles** [comment]
Back on Bluetooth ? 😍

**Comment by atrovato** [comment]
Just tried with last @abandonware/noble, but not working on windows...
And update branch to fix merge conflicts by the way.

File: .circleci/load_env.sh [hunk 2]
```diff
@@ -2,10 +2,10 @@
 
 echo 'export GITHUB_REPO=GladysAssistant/Gladys' >>$BASH_ENV
 echo 'export IMAGE=gladys' >>$BASH_ENV
-echo 'export REGISTRY=gladysassistant' >>$BASH_ENV
+echo 'export REGISTRY="${DOCKERHUB_REPO}"' >>$BASH_ENV
```
Comment (Pierre-Gilles) [inline .circleci/load_env.sh]: I'm not sure this is working. $DOCKERHUB_REPO is already used and is "gladysassistant/gladys", not "gladysassistant"

**Comment by Pierre-Gilles** [comment]
Thank you very much for this PR, this is gold 🥇 

I just put one small comment. 

Do you think it could be merged in master? I tested it on my side, it works fine on the UI/discovery part.

I don't have much devices to tests so I'm not the best user to test this, but if it's stable we can definitely merge it on master so all users can try it before the RC :)

File: .circleci/load_env.sh (hunk 2 above)
Comment (VonOx) [reply .circleci/load_env.sh]: You are right, i need to clean env and parameter in ci flow + add information in dev documentation

In that case IMAGE_ID is wrong

Comment (atrovato) [reply .circleci/load_env.sh]: So should I rollback for now, waiting @vonox to fix it in another PR?

Comment (VonOx) [reply .circleci/load_env.sh]: Yes I think Alex

Comment (atrovato) [reply .circleci/load_env.sh]: Ok, I'll rollback both modified in circleci folder (config.yml / load_env.sh).
@VonOx I hope you'll fix it ASAP :)

Comment (VonOx) [reply .circleci/load_env.sh]: Yep, I will do this week

**Comment by Pierre-Gilles** [comment]
Hey! Thanks for your changes. 

Is this ready to merge?

**Comment by LM1LC3N7** [comment]
🥳

**Comment by Pierre-Gilles** [comment]
Many thanks for your work on this major PR @atrovato ! 🥳 

This was the n°1 requests for Gladys 4, and we now have it :) 

Can't wait to see more integrations around the bluetooth in Gladys.

**Pull request merged**